src/options.rs

```diff
@@ -0,0 +1,117 @@
+use std::io;
+
+use serde::{de, ser};
+
+use crate::de::Deserializer;
+use crate::error::Result;
+use crate::extensions::Extensions;
+use crate::ser::{PrettyConfig, Serializer};
+
+pub struct Options {
+    default_extensions: Extensions,
+}
+
+impl Options {
+    #[must_use]
+    pub fn build() -> Self {
+        Self {
+            default_extensions: Extensions::empty(),
+        }
+    }
+
+    #[must_use]
+    pub fn with_default_extensions(mut self, default_extensions: Extensions) -> Self {
+        self.default_extensions = default_extensions;
```

Comment:
```suggestion
        self.default_extensions |= default_extensions;
```

Reply:
That would definitely allow for nicer code - do you think it would then also need a separate replacement / reset method?

Reply:
I'm not sure why that would be needed. Unless you got a specific case in mind, I wouldn't add that.

Reply:
Ok, thanks for your assessment. I didn't have anything in mind really, my usecase has a constant config that I only need to build once.

Reply:
If the extensions are just assigned, there's no need to make use of the builder pattern imo - I guess we could offer both, a builder and a "constructor".

Reply:
I agree that right now a builder is certainly overkill. But maybe additional config options will be added in the future, when having a constructor with lots of parameters would become annoying.